### DISCUSSION TITLE
Lesson 3: "send" and "transfer" are only available for objects of type "address payable", not "address".

### DISCUSSION THREAD

**Discussion by MartinSoka** [smartcontractkit/full-blockchain-solidity-course-py#1415] [Q&A] [answered]
Hi all, 
I'm following the tutorial with the latest compiler version 0.8.7. I get the following compiler error while implementing the withdraw function. 

TypeError: "send" and "transfer" are only available for objects of type "address payable", not "address".
 msg.sender.transfer(address(this).balance);

This was not needed with 0.6.x. Any ideas where can i find the compiler release notes/key changes where its described that i need to type cast the address ?

**Comment by cromewar** [comment, marked as answer]
Hello @MartinSoka the reason is because in solidity 0.8.x the payable identifier is not added automatically so the solution is this:
`msg.sender.transfer(payable(address(this).balance)`
Give it a try ;)

**Reply by MartinSoka** [reply]
Hey, thx for the reply.

Actually i used the following
`payable (msg.sender).transfer(address(this).balance);`
which worked on the end (not 100% sure if it was the correct approach tho). I also tried your suggestion but that one is not compiling (0.8.7).

But my question was more about the compiler change notes/docs, if i see other inconsistency in the future.

**Reply by cromewar** [reply]
I'm glad is solved now ;)
